packages/mds-geography-author/api.ts

```diff
@@ -0,0 +1,292 @@
+        if (details) {
+          throw new ValidationError(JSON.stringify(details))
+        }
+        await db.editGeography(geography)
+        return res.status(201).send(geography)
+      } catch (err) {
+        await log.error('failed to edit geography', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ result: 'not found' })
+        }
+        if (err instanceof ValidationError) {
+          return res.status(400).send({ error: err })
+        }
+        return res.status(500).send(new ServerError(err))
+      }
+    }
+  )
+
+  app.delete(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => scopes.includes('geographies:write')),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        await db.deleteGeographyMetadata(geography_id)
+        await db.deleteGeography(geography_id)
+        return res.status(200).send({ result: `Successfully deleted geography of id ${geography_id}` })
+      } catch (err) {
+        await log.error('failed to delete geography', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ error: err })
+        }
+        if (err instanceof AlreadyPublishedError) {
+          return res.status(405).send({ error: err })
+        }
+        return res.status(500).send(new ServerError())
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies/:geography_id/meta'),
+    checkAccess(scopes => {
+      return scopes.includes('geographies:read:published') || scopes.includes('geographies:read:unpublished')
+    }),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        const geography_metadata = await db.readSingleGeographyMetadata(geography_id)
+        const geography = await db.readSingleGeography(geography_id)
+        if (!geography.publish_date && !res.locals.scopes.includes('geographies:read:unpublished')) {
+          throw new InsufficientPermissionsError('permission to read metadata of unpublished geographies missing')
+        }
+        return res.status(200).send(geography_metadata)
+      } catch (err) {
+        await log.error('failed to read geography metadata', err.stack)
+        if (err instanceof NotFoundError) {
+          return res.status(404).send({ error: err })
+        }
+        if (err instanceof InsufficientPermissionsError) {
+          return res.status(403).send({ error: err })
+        }
+        return res.status(500).send({ error: new ServerError() })
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies'),
+    checkAccess(scopes => {
+      return scopes.includes('geographies:read:published') || scopes.includes('geographies:read:unpublished')
+    }),
+    async (req, res) => {
+      const summary = req.query.summary === 'true'
+      const { get_published = null, get_unpublished = null } = req.query
+      const params = { get_published, get_unpublished }
+      if (get_published) {
```

Comment:
I believe that `readGeographies()` does some validation such that both `get_published` and `get_unpublished` cannot both be set to `true`. Would it make sense to do the validation here, since this is where we parse the query params?

Reply:
You don't think that's kinda redundant? I'd rather throw just once in one layer.

Reply:
I agree that all the params validation should happen in one place ideally, I just thought maybe it would make sense to validate the query params coming in from the request before we get into the `db` method, but I'm fine with either approach.